docs/tutorials/2025_ieee_qw/04_rewriters.ipynb

```diff
@@ -142,130 +94,127 @@
    "id": "6c48966e",
    "metadata": {},
    "source": [
-    "The `rewriter` variable here is actually a `dataclass`, and in an interactive environment it displays a `KaTeX`-friendly expression so we can see the effect of method calls immediately.\n",
+    "The `rewriter` variable here is actually a `dataclass`, and in an interactive environment it displays a `KaTeX`-friendly expression so we can see the effect of method calls immediately."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "faf57c4b",
+   "metadata": {},
+   "source": [
+    "## Substitutions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dda1e137",
+   "metadata": {},
+   "source": [
+    "Rewriters support arbitrary substitutions of symbols or expressions, for example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bb5f758d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Here is an example of how to use the `substitute` method.\n",
+    "rotations.substitute(\"N_spatial - 1\", \"X\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "358d4e95",
+   "metadata": {},
+   "source": [
+    "All occurrences of $N_{spatial} - 1$ have been replaced with a new variable, $X$! Note that this change is **not permanent** as we didn't override the `rotations` variable.\n",
+    "\n",
+    "Notice that the term $\\max(b_{as}, b_{givens}, b_{mas})$ occurs a number of times in the `rotations` expression. \n",
+    "\n",
+    "One simplification we can make is to define a new variable: $$B:=\\max(b_{as}, b_{givens}, b_{mas}).$$ \n",
     "\n",
-    "Notice that the term $\\max(b_{as}, b_{givens}, b_{mas})$ occurs a number of times in this expression. For simplicity we can replace this with a new variable: $B:=\\max(b_{as}, b_{givens}, b_{mas})$."
+    "In the next cell, use the `.substitute` method to update the `rotations` variable with this change."
```

Comment:
But the next cell already does that, so there's nothing that the user needs to change, no?

Reply:
Same for the B > 5 cell later.